dll/win32/browseui/explorerband.cpp

```diff
@@ -261,6 +262,50 @@ CExplorerBand::NodeInfo* CExplorerBand::GetNodeInfo(HTREEITEM hItem)
     return reinterpret_cast<NodeInfo*>(tvItem.lParam);
 }
 
+static HRESULT GetCurrentLocationFromView(IShellView &View, PIDLIST_ABSOLUTE &pidl)
+{
+    CComPtr<IFolderView> pfv;
+    CComPtr<IShellFolder> psf;
+    HRESULT hr = View.QueryInterface(IID_PPV_ARG(IFolderView, &pfv));
+    if (SUCCEEDED(hr) && SUCCEEDED(hr = pfv->GetFolder(IID_PPV_ARG(IShellFolder, &psf))))
+        hr = SHELL_GetIDListFromObject(psf, &pidl);
+    return hr;
+}
+
+HRESULT CExplorerBand::GetCurrentLocation(PIDLIST_ABSOLUTE &pidl)
+{
+    pidl = NULL;
+    CComPtr<IShellBrowser> psb;
+    HRESULT hr = IUnknown_QueryService(m_pSite, SID_STopLevelBrowser, IID_PPV_ARG(IShellBrowser, &psb));
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    CComPtr<IBrowserService> pbs;
+    if (SUCCEEDED(hr = psb->QueryInterface(IID_PPV_ARG(IBrowserService, &pbs))))
+        if (SUCCEEDED(hr = pbs->GetPidl(&pidl)) && pidl)
+            return hr;
+
+    CComPtr<IShellView> psv;
+    if (!FAILED_UNEXPECTEDLY(hr = psb->QueryActiveShellView(&psv)))
+        if (SUCCEEDED(hr = psv.p ? GetCurrentLocationFromView(*psv.p, pidl) : E_FAIL))
```

Comment:
Please avoid assignations within if-conditions and macros. Also, multiline if-block bodies should have brackets (so in your case, the outermost if-block)

Reply:
I strongly disagree, I find this code to be pretty. Adding just one set of braces makes it ugly.

It's common COM pattern, QI and then do X.